### PR TITLE
feat: Add standard tracing points to portions of the mempool and consensus

### LINKED ISSUE #978: Add a default set of trace data collected
> I think that often these will be the same data points, thus by just capturing all metrics by default, we save having to rewrite this every time we want to analyse something.

originally posted in https://github.com/celestiaorg/celestia-core/pull/970#discussion_r1132148994 by @cmwaters 

In #970, we added the ability to collect arbitrary data into influx db, but didn't add any points in the code where we would actually collect data by default. We should add some default set of collection points.

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This PR updates the trace client with a slight refactor (renaming to be more consistent) and then adds a few standard tables to store traced information. We are also able to select which tables we wish to update (and therefore which information to trace) in the config.

closes #978

### CODE REVIEW

File: consensus/state.go [hunk 1]
```diff
@@ -142,7 +143,7 @@ type State struct {
 	// for reporting metrics
 	metrics *Metrics
 
-	eventCollector *trace.Client
+	traceClient *trace.Client
```
Comment (evan-forbes) [inline consensus/state.go:146]: apologies for not simply including the name change in a separate PR, it was driving me crazy enough that I just changed it here

**Comment by rootulp** [comment]
@evan-forbes golangci-lint is failing [here](https://github.com/celestiaorg/celestia-core/actions/runs/5786195593/job/15680419203?pr=1055#step:5:13377)

**Review by evan-forbes — COMMENTED** [review]
I'm not married to this design, so if other's have ideas I'm definitely open to them.

I've thought about refactoring the schema package to just be part of the trace package, and then make each `Write` function a method of the trace.Client. I've also though about moving each of the `Write` functions to their respective packages. This would allow us to have simpler `Write` functions. Instead of having to parse each piece of data we want to record in a trace like `WriteXYZ(client, x, y, x)`, we could simply pass the relevant struct and convert the struct in that function `WriteXYZ(client, XYZ). The downside being that we wouldn't have a single place that defines all of the schemas.

**Review by rootulp — DISMISSED** [review]
LGTM

[question] Would this tracing be valuable in upstream CometBFT? With the exception of the CAT pool specific tracing, the rest of this PR seems applicable upstream.

File: pkg/trace/schema/mempool.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		StateUpdateFieldKey:  stateUpdate,
+		VersionFieldKey:      version,
+	})
+}
+
+const (
+// LocalTable is the tracing "measurement" (aka table) for the local mempool
+// updates, such as when a tx is added or removed.
+// TODO: actually implement the local mempool tracing
+// LocalTable = "mempool_local"
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply pkg/trace/schema/mempool.go:121]: good idea, added https://github.com/celestiaorg/celestia-core/issues/1056

**Review by rootulp — APPROVED** [review]

File: pkg/trace/schema/tables.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package schema
+
+import "github.com/tendermint/tendermint/config"
+
+func init() {
+	config.DefaultInfluxTables = AllTables()
+}
+
+func AllTables() []string {
```
Comment (staheri14) [inline pkg/trace/schema/tables.go:9]: A func description would be helpful here; specifically, what does the string slice contain?

File: pkg/trace/schema/mempool.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+// Schema constants for the mempool_tx table
+const (
+	// MempoolTxTable is the tracing "measurement" (aka table) for the mempool
+	// that stores tracing data related to gossiping transactions.
+	//
+	// The schema for this table is:
+	// | time | peerID | tx size | tx hash | transfer type | mempool version |
+	MempoolTxTable = "mempool_tx"
+
+	// TxFieldKey is the tracing field key for receiving for sending a
```
Comment (staheri14) [inline pkg/trace/schema/mempool.go:27]: [nit] 
```suggestion
	// TxFieldKey is the tracing field key for receiving and sending a
```

**Review by staheri14 — APPROVED** [review]
Nice 👍

**Comment by evan-forbes** [comment]
merging now and moving onto #1056 where I'll a few more tables and the remaining feedback

**Pull request merged**